server/app/Icepeak.hs

```diff
@@ -40,30 +47,43 @@ installHandlers core serverThread =
     void $ installHandler Signals.sigINT
 
 
-readValue :: FilePath -> IO Value
-readValue filePath = do
+readValue :: FilePath -> Maybe Metrics.IcepeakMetrics -> IO Value
+readValue filePath metrics = do
   eitherEncodedValue <- try $ withFile filePath ReadMode hGetContents
   case (eitherEncodedValue :: Either SomeException ByteString) of
       Left exc -> error $ "Failed to read the data from disk: " ++ show exc
-      Right encodedValue -> case eitherDecodeStrict encodedValue of
-        Left msg  -> error $ "Failed to decode the initial data: " ++ show msg
-        Right value -> return value
-
+      Right encodedValue -> do
+        forM_ metrics $ Metrics.setDataSize (SBS.length encodedValue)
+        case eitherDecodeStrict encodedValue of
+          Left msg  -> error $ "Failed to decode the initial data: " ++ show msg
+          Right value -> return value
 
 main :: IO ()
 main = do
   env <- getEnvironment
   config <- execParser (configInfo env)
+
```

Comment:
This `main` function is growing places. I would be OK with it being long if the levels of abstraction it included were consistent, i.e, if it was just glueing things together, but there are some lower level stuff here that should be packed into a module IMO.

Reply:
I moved the log output for metrics to another place. I think overall, the main function is just gluing stuff together. First, it creates the stateful "objects", then it starts the thread, then it waits.

Loading the data file might be placed into the Core instead, but I think that is it out of the scope of this PR.

Reply:
Yeah, it is out of scope. But still, note that if `main` knows how to build some stateful "objects", it is not just glueing stuff.